### DISCUSSION TITLE
Issue with erode.m

### DISCUSSION THREAD

**Discussion by anamitrasikdar** [mtex-toolbox/mtex#2159] [Ask Anything] [answered]
Hi MTEX Community,

I am using erode function for cleanup, while on the second loop, each time it generates the following error. I have tried using the 
"small" mtexdata, as well. 
Inside each loop I am using fill option as well. Again no luck. It stops in the second loop. Is there something wrong with the ebsd.id?

I will appreciate any insight on this. Thanks
Anamitra

"Arrays have incompatible sizes for this operation.

Error in EBSD/subsind (line 84)
    ind = ind(:) & subs{i}(:);

Error in EBSD/subsref (line 34)
    ind = subsind(ebsd,s(1).subs);

Error in erode (line 92)
   ebsd = ebsd(~peroded);

Error in erode (line 82)
    ebsd = erode(ebsd,count,varargin{1}{i});

Related documentation"

**Comment by kilir** [comment]
Hi,

can you provide the relevant lines of code you use?

Cheers,

Rüdiger

**Comment by anamitrasikdar** [comment]
Sure,
for i=1:10
ebsd = erode(ebsd,2,{'n', 'P', 'O', 'Q'});
ebsd = ebsd.fill;
end

plot(ebsd); hold on
plot(ebsd('n'),'FaceColor','k'); hold off;
mtexTitle('no n, Q, P, O $\le$ 2 neighbors, filled,10 iter');

**Reply by kilir** [reply, marked as answer]
Hi,

for me this cf80f24 fixes the issue. Could you give it a try?

Cheers,
Rüdiger

**Reply by anamitrasikdar** [reply]
Hi, 
Unfortunately for my case it is showing something like this..

Arrays have incompatible sizes for this operation.

Error in EBSD/subsind (line 84)
    ind = ind(:) & subs{i}(:);

Error in EBSD/subsref (line 34)
    ind = subsind(ebsd,s(1).subs);

Error in erode (line 90)
   ebsd = ebsd(~peroded);

Related documentation

Thanks!
Anamitra

**Reply by kilir** [reply]
Can you provide the data which produces the error?

Cheers,

Rüdiger

**Reply by anamitrasikdar** [reply]
Sure! Here it is.
[test.zip](https://github.com/user-attachments/files/15926612/test.zip)

Regards,
Anamitra

**Reply by kilir** [reply]
Hi

I don't have any trouble with the data using the changes in the commit mentioned above.

```
load('test.mat')
for i=1:10
ebsd = erode(ebsd,2,{'n' 'P' 'O' 'Q'});
ebsd = ebsd.fill;
end
```

Cheers,

Rüdiger

**Reply by anamitrasikdar** [reply]
Hi Rüdiger,

You are correct. I probably omitted the "try" while applying the changes.
It works like a charm. Thank you.

Regards,
Anamitra
